lamindb/dev/db/_add.py

```diff
@@ -81,28 +81,84 @@ def add(  # type: ignore
             return results
         else:
             records = [model(**fields)]
-    for record in records:
-        if isinstance(record, DObject) and hasattr(record, "_local_filepath"):
-            upload_data_object(record, use_fsspec=use_fsspec)
+
     if session is None:  # assume global session
         session = settings.instance.session()
         settings.instance._cloud_sqlite_locker.lock()
         close = True
     else:
         close = False
+
+    added_records = []
+    raise_error = False
     for record in records:
-        session.add(record)
-    session.commit()
-    for record in records:
+        # commit metadata to database
+        prepare_filekey_metadata(record)
+        try:
+            session.add(record)
+            session.commit()
+            session.refresh(record)
+        except Exception:
+            raise_error = True
```

Comment:
Can you track the errors (or error messages), here?

So, you do something like
```
except Exception as e:
    raised_errors.append(e)
 ```

Reply:
It's going to very hard to understand for the user what went wrong otherwise.